base/MAPL_CapOptions.F90

```diff
@@ -80,5 +88,97 @@ function new_CapOptions(unusable, cap_rc_file, egress_file, ensemble_subdir_pref
 
    end function
 
+#ifdef USE_FLAP
```

Comment:
I made a change and kept CapOptions as a data struct by moving set_esmf_looging_mode into the constructor. That makes sense because only Flap knows about flag_name.  I did have another implementation that moves  "constructors" to FlapCLI,  the GEOSgcm main would look like this
cli = MAPL_FlapCLI(.....)
cap_options = cli%get_mapl_cap_options(...)
cap = MAPL_Cap(...)

But I think this is not as good as the one now we have